lib/foreman_maintain/runner.rb

```diff
@@ -84,6 +86,14 @@ def storage
 
     private
 
+    def execute_scenario_steps(scenario)
+      scenario.before_scenarios.flatten.each { |before_scenario| run_scenario(before_scenario) }
+      yield
```

Comment:
I think we can also move `return if quit? # the before scenarios caused the stop of the execution` here and not use the `yield`, for readability purposes

Reply:
Yes, I thought returning from extracted-method might cause some unexpected behaviour.